core/src/test/java/org/apache/iceberg/TestTables.java

```diff
@@ -255,8 +257,13 @@ void failCommits(int numFailures) {
       this.failCommits = numFailures;
     }
 
+    int getMetadataFetchCount() {
```

Comment:
Decided to count calls to current()/refresh() as a proxy for reading metadata from objectstorage. For TestTables (which uses in memory metadata) that is the best we can do.

Is there some other test I could write that uses real on file system metadata (not TestTables)?

Reply:
Why current/refresh counting is not great: cached metadata fetches are free while uncached fetches are not, and this does not differentiate between the two.

Instrumenting at the fileio level would let us validate indeed only a single uncached metadata request is needed for the whole commit.